File: Tarea 4 - La redencion/terror.py
#Juan Miguel Rojas Mejía
#8963761

from sys import stdin

def solve(N, K, X, Y):
    ans = 0

    for i in range(N):
        Y[i] = Y[i] - X[i]
    
    Y.sort()

    for i in range(N):
        if(Y[i] < 0 and K != 0):
            K -= 1
        else:
            ans += Y[i]
    
    return ans

def main():
    C = int(stdin.readline())
    iter = 0
    while(C != iter):
        N, K = map(int, stdin.readline().split())
        X = list(map(int, stdin.readline().split()))
        Y = list(map(int, stdin.readline().split()))
        ans = solve(N, K, X, Y)
        if(ans <= 0):
            print(f"Case {iter+1}: No Profit")
        else:
            print(f"Case {iter+1}: {ans}")
        iter+=1

main()

File: Tarea 4 - La redencion/terror.cpp
#include <iostream>
#include <fstream>
#include <vector>
#include <algorithm>

using namespace std;

int solve(int N, int K, vector<int>& X, vector<int>& Y) {
    int ans = 0;
    for (int i = 0; i < N; i++) 
        Y[i] = Y[i] - X[i];
    sort(Y.begin(), Y.end());

    for (int j = 0; j < N; j++) {
        if (Y[j] < 0 && K != 0) {
            K--;
        } else {
            ans += Y[j];
        }
    }
    
    return ans;
}

int main() {
    //ifstream cin("terrorin.txt");
    //ofstream cout("terrorout.txt");

    int C, N, K, ans;
    cin >> C;
    int iter = 0;
    while (C--) {
        cin >> N >> K;
        vector<int> X(N);
        vector<int> Y(N);
        for (int i = 0; i < N; i++) cin >> X[i]; 
        for (int j = 0; j < N; j++) cin >> Y[j];
        ans = solve(N, K, X, Y);
        if (ans <= 0) {
            cout << "Case " << iter + 1 << ": No Profit" << endl;
        } else {
            cout << "Case " << iter + 1 << ": " << ans << endl;
        }
        iter++;
    }

    //cin.close();
    //cout.close();

    return 0;
}
